apps/explorer/src/components/module/PkgModulesWrapper.tsx

```diff
@@ -48,14 +48,30 @@ function PkgModuleViewWrapper({ id, modules }: Props) {
     const [searchParams, setSearchParams] = useSearchParams();
     const [query, setQuery] = useState('');
 
-    const paramModule = searchParams.get('module') || modulenames?.[0] || null;
-
-    const [selectedModule, setSelectedModule] = useState(
-        !!paramModule && modulenames.includes(paramModule)
-            ? paramModule
-            : modulenames[0]
+    const convertedSearchParams = useMemo(
```

Comment:
Search params should already be a search params object, you shouldn’t need to convert it here.

Reply:
I notice you use this below, you should just re-create this inline like you did before. The point of this is actually just to clone the search params before modifying them, so we want to do that inline